code/game/objects/structures/watercloset.dm

```diff
@@ -472,3 +466,31 @@
 	icon_state = "bounty-open"
 	color = null
 	alpha = 255
+
+/obj/structure/curtain/proc/toggle(mob/M)
+    if (check(M))
+        open = !open
+        playsound(loc, 'sound/effects/curtain.ogg', 50, 1)
+        update_appearance()
+    return
+
+/obj/structure/curtain/proc/check(mob/M)
+    if (istype(src, /obj/structure/curtain/directional))
+        if (src.dir != get_dir(src, M))
+            return TRUE
+    else
+        return TRUE
+    return FALSE
```

Comment:
Rather than checking the type of src, override the proc by adding a new function `/obj/structure/curtain/directional/check(mob/M)` This will call that function instead of this one when the type of src is `/obj/structure/curtain/directional`.